src/app/feed/info-box/info-box.component.ts

```diff
@@ -42,19 +42,30 @@ export class InfoBoxComponent implements OnInit, OnChanges {
 		this.parseApiResponseData();
 	}
 	sortHashtags(statistics) {
-		let sortable = [];
 		/* A check for both the data and the individual objects is necessary, also if the data is not empty*/
+		let stored = [];
 		if (statistics !== undefined && statistics.length !== 0) {
 			for (const s in statistics) {
 				if (s) {
-					sortable.push([s, statistics[s]]);
+					for (let i = 0; i < statistics[s].length; i++) {
+						stored.push(statistics[s][i]);
+					}
 				}
 			}
-			sortable.sort(function (a, b) {
-				return b[1] - a[1];
-			});
-			sortable = (sortable.slice(0, 10));
-			this.topHashtags = sortable;
+			stored = stored.reduce(function (acc, curr) {
+				if (typeof acc[curr] === 'undefined') {
+						acc[curr] = 1;
+				} else {
+					acc[curr] += 1;
+				}
+				return acc;
+			}, []);
+			this.topHashtags = Object.keys(stored)
+			.map(key => key.trim())
+			.filter(key => key !== '')
+			.map(key => ([key, stored[key]]))
+			.sort((a, b) => b[1] - a[1]);
+			this.topHashtags = this.topHashtags.slice(0, 10);
```

Comment:
This is kind of repetitive, you can define a function do to this.

Reply:
Yes, but there are two patches, one/both needs to be merged to make it feasible.